packages/@ember/-internals/glimmer/lib/component-managers/custom.ts

```diff
@@ -149,16 +162,82 @@ export default class CustomComponentManager<ComponentInstance>
     const { delegate } = definition;
     const capturedArgs = args.capture();
 
-    const component = delegate.createComponent(
-      definition.ComponentClass.class,
-      capturedArgs.value()
-    );
+    let value;
+    let namedArgsProxy = {};
+
+    if (EMBER_CUSTOM_COMPONENT_ARG_PROXY) {
+      if (HAS_NATIVE_PROXY) {
+        let handler: ProxyHandler<{}> = {
+          get(_target, prop) {
+            assert('args can only be strings', typeof prop === 'string');
+
+            let tracker = getCurrentTracker();
+            let ref = capturedArgs.named.get(prop as string);
+
+            if (tracker) {
+              tracker.add(ref.tag);
+            }
+
+            return ref.value();
+          },
+        };
+
+        if (DEBUG) {
+          handler.set = function(target, prop) {
+            assert(
+              `You attempted to set ${target}#${String(
```

Comment:
We chatted a bit about this, but I'd like to get the component itself into this message `target` is going to be the empty `{}` POJO you make just above. Can we update this to `definition.ComponentClass.class` instead? At least it would toString the component name...